### ISSUE TITLE
Ratings cannot be submitted without JS

### ISSUE THREAD

**Issue by tofumatt** [mozilla/addons-frontend#1552]
### Describe the problem and steps to reproduce it:
Rating buttons submit to `<form action="">` when JS isn't available. See: https://github.com/mozilla/addons-frontend/blob/master/src/amo/components/OverallRating.js#L102

This means the ratings can't be submitted without JS.

### What happened?
I clicked a button and the page submitted to itself, so it seemed like nothing happened.

### What did you expect to happen?
I click the button to rate the add-on and the add-on is shown as rated by me with the appropriate amount of stars.

### Anything else we should know?
I encountered this in development mode and maybe the buttons actually show content elsewhere... as none of my CSS displayed on the development server without JS, but that could be separate.

### Screenshots
<img width="316" alt="screenshot 2017-01-01 13 05 59" src="https://cloud.githubusercontent.com/assets/90871/21581288/87a9a82c-d023-11e6-9a6b-9ef89d3a2fa3.png">

**Comment by muffinresearch** [comment]
Write requests with data are not going to result in a server-side response with updated UI presently. 

This is a big missing piece in the universal app world currently. Not sure if anyone's been working on solving this elsewhere but it would be great to fix this omission. Because then *everything* would work universally.

With GET's (e.g. search) you will have a response for free since it's just a URL, but with a POST/PUT etc we'd need to have the server handle the data sent, forward it onto the API, get the response and then put things back together and send it to the client. 

This is kind of what we're already doing for other GETs we just need to patch in the piece that hands off the data to the API from the server ahead of the render whereas currently we're just always requesting data. The only other bits that will make this more involved are ensuring authentication gets delegated when the server makes the requests on behalf of the client. But assuming cookies maybe that bit's not too bad.

One other piece to this is differentiating the URL for the form action to trigger the server-side-only behaviour. I had an idea that the form action could point to a slightly different url pattern to trigger the server-side render and then we would just be overriding the submits when JS is present for the default when the client is directly talking to the API.

I don't think this is necessary for V1 but if we could find a nice non-messy way to achieve this it would be a great addition to the project.

**Comment by tofumatt** [comment]
Aha, this makes total sense once I stop to think about it, I didn't get beyond "we don't have an `action="/new-review"` or whatnot".

Shame, but makes total sense 😢

**Comment by kumar303** [comment]
maybe we should change it to `action="/javascript-must-be-enabled"` so that they land on a more sensical page. 

I don't think we'll ever really need to support posting forms without javascript. The universal use case for us is more about SEO (just GETs). There are plenty of accessibility solutions that deal with with JS already.

**Comment by tofumatt** [comment]
Oh, that's a really cool idea. I'm for that 👍🏻

- tofumatt

**Comment by muffinresearch** [comment]
@kumar303 that's a great idea that solves the immediate need!

The longer term case for being able to interact via forms without JS is more about redundancy. There are a lot of reasons why JS can not be available and I know we all know this but it's a useful reminder [1]. 

So I still think it's a problem worth solving (if we can solve it generically without a ton of special casing) but it's certainly a lot lower priority than other things we've got going on.

[1] http://kryogenix.org/code/browser/everyonehasjs.html

**Comment by muffinresearch** [comment]
Closing this one in favour of https://github.com/mozilla/addons-frontend/issues/4260
